src/e3/anod/deps.py

```diff
@@ -79,8 +79,24 @@ def __init__(
         self.host = host
         self.target = target
         self.build = build
-        self.qualifier = qualifier
         self.local_name = local_name if local_name is not None else name
+
+        self.qualifier: str | None
+        if isinstance(qualifier, dict):
+            for q in set(qualifier.keys()):
+                v = qualifier[q]
+                if isinstance(v, bool):
+                    # It's a tag qualifier
+                    if v:
+                        qualifier[q] = ""
+                    else:
+                        qualifier.pop(q)
+            self.qualifier = ",".join(
```

Comment:
sort by key and ensure that list, set or frozenset are ordered too

Reply:
Done